kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        except FileNotFoundError:
+            config_data = {}
+
+        return config_data
+
+
+    def _write_config(self, config_data, indent=1):
+        '''write config data to file.
+        '''
+        with open(self.config, 'w') as f:
+            json.dump(config_data, f, indent=indent)
+
+
+    def set_config_value(self, name, value, quiet=False):
+        '''a client helper function to set a configuration value, meaning
+           reading in the configuration file (if it exists), saving a new
+           config value, and then writing back
+        '''
+
+        config_data = self._read_config()
+   
+        # If defined by client, set and save!
+        self._write_config(config_data)
+
+        if value is not None:
+
+            # Update the config file with the value
+            config_data[name] = value
+
+            # Update the instance with the value
+            self.config_values[name] = value
+
+            if not quiet:
+                self.print_config_value(name, separator=' is now set to: ')
+
 
     def unset_config_value(self, name, quiet=False):
-        try:
-            with open(self.config, 'r') as f:
-                config_data = json.load(f)
-            config_data[name] = None
-            with open(self.config, 'w') as f:
-                json.dump(config_data, f, indent=2)
-            self.config_values[name] = None
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        '''unset a configuration value'''
 
-    def copy_config_value(self, name, values):
-        if name in values:
-            self.config_values[name] = values[name]
+        config_data = self._read_config()
+
+        # Remove it, if exists, both from loaded file and client
+
+        del self.config_values[name] # is there reason this was None?
+
+        if name in config_data:
+            del config_data[name]
+
+            self._write_config(config_data)
+
+            if not quiet:
+                self.print_config_value(name, separator=' is now set to: ')
 
     def get_config_value(self, name):
+        ''' return a config value (with key name) if it's in the config_values,
+            otherwise return None
+ 
+            Parameters
+            ==========
+            name: the config value key to get
+
+        '''
         if name in self.config_values:
             return self.config_values[name]
-        return None
 
-    def get_default_download_dir(self, *subdirs):
+    def get_config_path(self):
+        '''get the path configuration value, otherwise return default.
+        '''
         path = self.get_config_value(self.CONFIG_NAME_PATH)
         if path is None:
-            return os.getcwd()
-        else:
-            return os.path.join(path, *subdirs)
+            return self.config_path
```

Comment:
Where is `self.config_path` set?  Additionally, it is very important that this defaults to `os.getcwd()` when downloading files.  Right now, I don't think your implementation does that.  As noted just above, `get_config_directory` and `get_default_download_directory` are distinct operations.

Reply:
> Where is self.config_path set?

Right now this is set upon initialization (I don't think I wrote this bit, and I thought about changing it because it tries to make a directory on init and I wouldn't advise that). 

```python
    config_dir = os.environ.get('KAGGLE_CONFIG_DIR') or os.path.join(
        expanduser('~'), '.kaggle')
    if not os.path.exists(config_dir):
        os.makedirs(config_dir)
```

> Additionally, it is very important that this defaults to os.getcwd() when downloading files.

Ah! I think I understand what you were saying above about distinguishing the two. I now see where / how the function is used to download, and what I'll do is to separate them by still setting the config_dir upon init, but then for that function changing the default to os.getcwd(). So they are separate. 

I've done these changes and will push in a second, still reading!